### DISCUSSION TITLE
Connection using service principal/app registration for PowerBI

### DISCUSSION THREAD

**Discussion by pmatthews05** [microsoft/finops-toolkit#699] [Q&A] [answered]
Hi, I'm looking the PowerBI finops template, and I can see when I open it to grab our Enterprise Agreement data it is using my account. Is there a way to make it use a service principal? 

I'm unable to use a service account, and I'm not allowed to use my account to gather the data.

**Comment by flanakin** [comment, marked as answer]
@pmatthews05 Please see https://learn.microsoft.com/azure/cost-management-billing/manage/assign-roles-azure-service-principals

**Reply by pmatthews05** [reply]
Hi @flanakin 
I'm aware that you can connect using a Service Principal with API calls, however the PowerBI template when it requests a way to sign in, there doesn't appear to be a way to do this via a Service Principal. Only by my own account. It's here I would like the option to connect with Service Principal.

<img width="542" alt="image" src="https://github.com/microsoft/finops-toolkit/assets/6736509/f7fcdf2b-4c67-44b9-a4ba-6d360fca7718">

I think I have found a way around our problem since I posted this question. I'm using FinOps hubs. This uses the Cost exports into a storage account and the PowerBI template there you can connect to the Storage account using a SAS or Storage Key. Therefore my account isn't required, and I'm not using a service account to pull the data for each user, using the PowerBI workbook.

**Reply by flanakin** [reply]
Yes, that's ultimately where we plan to go. We just haven't got there yet. I'm hoping to have support for more datasets in the coming months...
